truffle/com.oracle.truffle.api.instrumentation/src/com/oracle/truffle/api/instrumentation/ProbeNode.java

```diff
@@ -280,6 +282,8 @@ final void onDispose(EventContext context, VirtualFrame frame) {
         final void onEnter(EventContext context, VirtualFrame frame) {
             try {
                 innerOnEnter(context, frame);
+            } catch (KillException | QuitException ex) {
+                throw ex;
             } catch (Throwable t) {
```

Comment:
Can we implement it like this? I don't think it makes every sense to have Kill on the fastpath. Also we should merge the control flow edge with other exceptions in case of langauge bindings.

```
 static void failEventForInstrumentation(EventBinding<?> b, String eventName, Throwable t) {
        if (t instanceof KillException) {
            throw (KillException) t;
        }
  ...
}

```